integration-cli/docker_cli_push_test.go

```diff
@@ -292,8 +308,15 @@ func (s *DockerRegistryAuthTokenSuite) TestPushMisconfiguredTokenServiceResponse
 	// TODO: isolate test so that it can be guaranteed that the 503 will trigger xfer retries
 	// assert.Assert(c, strings.Contains(out, "Retrying"))
 	// assert.Assert(c, !strings.Contains(out, "Retrying in 15"))
-	split := strings.Split(out, "\n")
-	assert.Equal(c, split[len(split)-2], "toomanyrequests: out of tokens")
+
+	// Auth service errors are not part of the spec and containerd doesn't parse them.
+	if testEnv.UsingSnapshotter() {
+		assert.Check(c, is.Contains(out, "failed to authorize: failed to fetch anonymous token"))
+		assert.Check(c, is.Contains(out, "503 Service Unavailable"))
```

Comment:
We're using our test server for this, but curious if it reflects what docker hub returns; https://github.com/moby/moby/blob/c8b9dfb25e22a020c6828bbc5b87d2956b2a038e/integration-cli/docker_cli_push_test.go#L238-L246

Reply:
Yes, because that test server returns `TOOMANYREQUESTS` after the 3 tries (which give `UNAVAILABLE`).
Containerd doesn't do the retry and it fails fast with the `UNAVAILABLE` and doesn't reach the `TOOMANYREQUESTS`.

Reply:
Wondering now if the retry was added to account for "cron-jobs" (I recall some threads about some hours being notable for cron-jobs hammering Docker Hub, which may cause brief moments where services can return a 503).